src/CSLOLUtils.cpp

```diff
@@ -118,10 +118,84 @@ bool CSLOLUtils::isUnnecessaryAdmin() {
     }
     return result;
 }
-#else
-// TODO: macos implementation
-QString CSLOLUtils::detectGamePath() { return ""; }
+#elif __APPLE__
+#    include <libproc.h>
+#    include <string.h>
+
+#    include <filesystem>
+#    include <iostream>
+
+QString CSLOLUtils::detectGamePath() {
+    pid_t *pid_list;
+
+    //
+    //  Get necessary size for `pid_list`
+    //
+    int bytes = proc_listallpids(nullptr, 0);
+
+    //
+    //  Something went wrong trying
+    //  to find all the processes.
+    //
+    if (bytes <= 0) {
+        return "";
+    }
+
+    pid_list = (pid_t *)malloc(bytes * sizeof(pid_t));
+
+    bytes = proc_listallpids(pid_list, bytes * sizeof(pid_t));
+
+    for (int i = 0; i < bytes / sizeof(pid_list[0]); i++) {
```

Comment:
```suggestion
    for (int i = 0; i < bytes; i++) {
```
From what i see proc_listallpids returns a count of items not size of bytes.

Reply:
That indeed appears to be the case as per the following snippet:

```cpp
int 
proc_listallpids(void * buffer, int buffersize)
{
	int numpids;
	numpids = proc_listpids(PROC_ALL_PIDS, (uint32_t)0, buffer, buffersize);

	if (numpids == -1)
		return(-1);
	else
		return(numpids/sizeof(int));
}
```

Source: <https://opensource.apple.com/source/xnu/xnu-2782.40.9/libsyscall/wrappers/libproc/libproc.c>

Reply:
Might make sense to rename the return variable to something more appropriate than "bytes"?

Reply:
Yep, check my comment in the conversation below this one.